dev/PushNotifications/PushNotificationsLongRunningTask.ProxyStub/NotificationsLongRunningProcess.idl

```diff
@@ -6,12 +6,25 @@ import "ocidl.idl";
 
 #include "../PushNotifications-Constants.h"
 
+[object]
+[uuid(25604D55-9B17-426F-9D67-2B11B3A65598)]
+[pointer_default(unique)]
+interface IWpnForegroundSink : IUnknown
+{
+    HRESULT InvokeAll([in] ULONG length, [in, string] byte data[]);
+};
+
 [object]
 [uuid(60FC21B2-B396-4D49-94F0-7555869FB93C)]
 [pointer_default(unique)]
 interface INotificationsLongRunningPlatform : IUnknown
 {
     HRESULT RegisterFullTrustApplication([in] LPCWSTR processName, [in] GUID remoteId, [out] GUID* appId);
+
+    HRESULT RegisterForegroundActivator([in] IWpnForegroundSink* sink, [in] LPCWSTR processName);
+
+    HRESULT SendBackgroundNotification([in] LPCWSTR processName, [in, string] byte data[], [in] ULONG length);
```

Comment:
What is this for? I don't get it.

Reply:
Ah just read through your code. This is if the developer intercepts the payload in the FG but for some strange reason also chooses to trigger the background activator?

Reply:
Yes. This is to simulate .Handled().

Reply:
Hmm why don't you have IWpnSink::InvokeAll return a specific error code if Handled == false. Then you could eliminate this API and handle everything from LRP side.

Reply:
Or InvokeAll could have an additional parameter out bool handled?

Reply:
Does the background handler get triggered once the .Handled(bool) is called?

Reply:
If the developer intercepts the payload in the foreground and sets args.Handled = true, then the background handler will not get invoked. Otherwise the background handler would also get invoked with the same payload.